services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.controller.ts

```diff
@@ -1,116 +1,324 @@
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiParam({
+    name: 'name',
+    required: true,
+    type: 'string',
+  })
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description:
+      'The Financial Service Provider Configuration property has been successfully updated.',
+  })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Bad request. Body or params are malformed',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description:
+      'Program does not exist or Financial Service Provider Configuration or propery does not exist',
+  })
+  @Patch(
+    ':programId/financial-service-provider-configurations/:name/properties/:propertyName',
+  )
+  public async updateProperty(
+    @Body()
+    property: UpdateProgramFinancialServiceProviderConfigurationPropertyDto,
+    @Param('programId', ParseIntPipe)
+    programId: number,
+    @Param('name')
+    name: string,
+    @Param('propertyName')
+    propertyName: WrapperType<FinancialServiceProviderConfigurationProperties>,
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponsePropertyDto> {
+    return await this.programFinancialServiceProviderConfigurationsService.updateProperty(
+      {
+        programId,
+        programFspConfigurationName: name,
+        propertyName,
+        property,
+      },
+    );
+  }
+
+  @AuthenticatedUser({ isAdmin: true })
+  @ApiOperation({
+    summary: `Delete a single Program Financial Service Provider Configuration property. See ${EXTERNAL_API.baseApiUrl}/financial-service-providers for required properties per financial service provider.`,
+  })
+  @ApiParam({ name: 'programId', required: true, type: 'integer' })
+  @ApiParam({
+    name: 'name',
+    required: true,
+    type: 'string',
+  })
+  @ApiParam({
+    name: 'propertyName',
+    required: true,
+    type: 'string',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description:
+      'The Financial Service Provider Configuration property is successfully deleted.',
+  })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Bad request. Body or params are malformed',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description:
+      'Program does not exist or Financial Service Provider Configuration or propery does not exist',
+  })
+  @Delete(
+    ':programId/financial-service-provider-configurations/:name/properties/:propertyName',
+  )
+  public async deleteProperty(
+    @Param('programId', ParseIntPipe)
+    programId: number,
+    @Param('name')
+    name: string,
+    @Param('propertyName')
+    propertyName: WrapperType<FinancialServiceProviderConfigurationProperties>,
+  ): Promise<void> {
+    return await this.programFinancialServiceProviderConfigurationsService.deleteProperty(
```

Comment:
is there a point to doing return if return type is void? (check similar cases in this file)